vnext/Shared/BaseScriptStoreImpl.cpp

```diff
@@ -139,22 +144,7 @@ std::unique_ptr<const jsi::Buffer> LocalFileSimpleBufferStore::getBuffer(const s
     std::terminate();
   }
 
-  // Treat buffer id as the relative path fragment.
-  std::ifstream file(storeDirectory_ + bufferId, std::ios::binary | std::ios::ate);
-
-  if (!file) {
-    return nullptr;
-  }
-
-  std::streamsize size = file.tellg();
-  file.seekg(0, std::ios::beg);
-
-  auto buffer = std::make_unique<ByteArrayBuffer>(static_cast<size_t>(size));
-  if (!file.read(reinterpret_cast<char *>(buffer->data()), size)) {
-    return nullptr;
-  }
-
-  return buffer;
+  return Microsoft::JSI::MakeMemoryMappedBuffer(winrt::to_hstring(storeDirectory_ + bufferId).c_str());
```

Comment:
>winrt::to_hstring [](http://example.com/codeflow?start=48&length=17)

Does this method perform proper unicode conversion? I believe that storeDirectory and bufferId can both contain UTF-8 characters?

Reply:
to_hstring should do proper UTF-8 to UTF-16 conversion.

Reply:
Hum I can't find any documentation that specifies that. @NickGerleman do you know where I can find it?

Also, as a separate issue, if to_hstring does proper unicode conversions, should we replace usages of the unicode conversion functions in Common with it?

Reply:
@hansenyy it does the proper conversion: you can see the source code for it in the base.h. Let's not change our code to use it yet.

Reply:
Yes, we should. I have done that in a few modules.
Regarding the docs, I infer the function does proper UTF conversion from the following:

[`winrt::to_string`](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/to-string) uses [WideCharToMultiByte](https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-widechartomultibyte) which seems to ensure UTF conversion.\
It's therefore very likely that [`winrt::to_hstring`](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/to-hstring) uses [MultiByteToWideChar](https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar), which should ensure proper UTF-8 to UTF-16 conversion.

This is also suggested by @kennykerr [here](https://kennykerr.ca/2018/02/28/cppwinrt-strings/).

I'll ping the C++/WinRT before merging.

Reply:
Yep, those helpers properly handle utf conversions back and forth.